src/nvim/ex_docmd.c

```diff
@@ -9625,20 +9625,34 @@ static void ex_folddo(exarg_T *eap)
 static void ex_terminal(exarg_T *eap)
 {
   char *name = (char *)p_sh;  // Default to 'shell' if {cmd} is not given.
```

Comment:
`name` and the `xfree(name);` at the end can be moved into the first `if` branch.